File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;

namespace dxWinSample {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            XtraReport report = GetReport();
            new ReportPrintTool(report).ShowRibbonPreview();
        }
        private XtraReport GetReport() {
            XtraReport mainReport = new XtraReportMain();
            mainReport.CreateDocument();
            XtraReport infoReport = new XtraReportInformation(mainReport.Pages.Count);
            infoReport.CreateDocument();
            int minPageCount = Math.Min(mainReport.Pages.Count, infoReport.Pages.Count);
            for(int i = 0; i < minPageCount; i++) {
                mainReport.Pages.Insert(i * 2 + 1, infoReport.Pages[i]);
            }
            return mainReport;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraReports.UI

Namespace dxWinSample
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim report As XtraReport = GetReport()
            CType(New ReportPrintTool(report), ReportPrintTool).ShowRibbonPreview()
        End Sub
        Private Function GetReport() As XtraReport
            Dim mainReport As XtraReport = New XtraReportMain()
            mainReport.CreateDocument()
            Dim infoReport As XtraReport = New XtraReportInformation(mainReport.Pages.Count)
            infoReport.CreateDocument()
            Dim minPageCount As Integer = Math.Min(mainReport.Pages.Count, infoReport.Pages.Count)
            For i As Integer = 0 To minPageCount - 1
                mainReport.Pages.Insert(i * 2 + 1, infoReport.Pages(i))
            Next i
            Return mainReport
        End Function
    End Class
End Namespace
